pkg/kubelet/config/common.go

```diff
@@ -59,7 +58,8 @@ func applyDefaults(pod *api.Pod, source string, isFile bool, nodeName types.Node
 			fmt.Fprintf(hasher, "url:%s", source)
 		}
 		hash.DeepHashObject(hasher, pod)
-		pod.UID = types.UID(hex.EncodeToString(hasher.Sum(nil)[0:]))
+		hexVlaue := hasher.Sum(nil)[0:]
```

Comment:
this function is commonly used. any reason this only occurs to Arktos ?

Reply:
Arktos made changes to ./staging/src/k8s.io/apiserver/pkg/registry/rest/meta.go and reused meta.uid instead of generate a new one (Alkaid always generate new uid)

Reply:
I think Sindica's comment makes sense.

Reply:
According to the code logics in this function, multiple static pods will have same UID if they are on the same node.  Will this be a problem?  Shall we just call GUID generation function to generate a unique UID?

(Previous this probably isn't a problem because API server will replace the UID, as Ying mentioned in the comment)

Reply:
I think more discussion is needed here:  1) Shall we keep the original behavior that API Server re-generates UID?  2) If we don't keep the original behavior and rely on client side to generate guid, shall we just call GUID generation function here?  I can see there is a problem here, multiple static pods will have same UID if they are on the same node.